oeps/oep-0007.rst

```diff
@@ -0,0 +1,526 @@
+Python 3 office hours to help answer questions and troubleshoot problems that
+arise during migration.
+
+Appropriate tooling will help.  Tests should be configured to run under both
+Python 2 or 3 using tox.  As it would double the cost of running
+tests to run both all the time, we may only want to enable Python 3 tests in
+the CI environment for repos that are actively being migrated.  We still want
+to ensure that Python 3 compatibility is improving on other projects, so a
+lightweight metric that can be run every time would also be useful, such as
+running pylint with the ``caniusepython3.pylint_checker`` extension.  Making these checks
+mandatory in a similar way to our current quality will ensure that
+compatibility is improving (TODO: timeline for this?).
+
+
+Migrating projects
+++++++++++++++++++
+
+We should be able to migrate individual applications to Python 3 independently.
+To begin with, we should pilot the process using a relatively small (but
+complex enough to provide useful information) IDA.  As we go, we will document
+the process, find pain points, figure out ways of dealing with them, and
+continue to improve our process.
+
+For a given project, steps are:
+
+1. Turn on caniusepy3k linting, and reset the lint error cap.
+2. Turn on tox testing in Python 3, but allow the tests to fail.
+3. Reduce the number of lint errors to zero, lowering lint error cap as you go.
+   Optionally, use ``futurize``, phase 1 to automate the first stage of the
+   conversion process.
+4. Reduce the number of failing tests to zero.  This may involve updating
+   dependent libraries to Python 3 compatibile versions.  It will almost
+   certainly involve normalizing text handling.
+5. Making failing Python 3 tests fail the build.
+6. Deploy the project in Python 3.
+7. Stop testing under Python 2.
+
+
+Order of migrations
++++++++++++++++++++
+
+* IDAs that we want to continue supporting in the future
+
+  * Old IDAs (that we want to replace) should not be upgraded, but we will need
+    to prioritize replacement to occur during the migration timeframe.
+
+* Implement remote execution of xBlocks (to allow a window of bicompatibility
+  for external xblocks)
+* edx-platform
+
+  * Deploy xblocks separately to test remote execution
+  * Migrate to Python 3
+  * Upgrade external xblocks as needed, and support partners who wish to do the
+    same
+
+Support libraries should be migrated as required by our migration schedule for
+the services that require them.  If external libraries need minor updates to
+support Python 3 that we can perform, we should opt to push those changes
+upstream rather than forking projects when possible.
+
+Code conversion should be automated as much as possible.  The future_ library
+includes a ``futurize`` executable that will do much of the legwork.  As we gain
+experience migrating code, we will develop a sense as to how aggressively we
+can use ``futurize``, and what other work needs to be done.
+
+
+Deploying Services
```

Comment:
I think you could remove this section. This is true of all significant changes like this and only applicable to edx.org